src/lib/KeyLib.sol

```diff
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.23;
+
+/// @dev The type of key.
+enum KeyType {
+    P256,
+    WebAuthnP256,
+    Secp256k1
+}
+
+struct Key {
+    /// @dev Unix timestamp at which the key expires (0 = never).
+    uint40 expiry;
+    /// @dev Type of key. See the {KeyType} enum.
+    KeyType keyType;
+    /// @dev Whether the key is a super admin key.
+    /// Super admin keys are allowed to call into super admin functions such as
+    /// `authorize` and `revoke` via `execute`.
+    bool isSuperAdmin;
```

Comment:
i think we can keep isSuperAdmin for now, but let's remove this comment? I think we can just allow superAdmin keys to execute any call

Reply:
execute any external call

Reply:
sg!